File: lib/aiagentskills_site_kit.dart
const baseUrl = 'https://aiagentskills.net';

String _cleanSlug(String slug) => slug.trim().replaceAll(RegExp(r'^/+|/+$'), '');
String urlFor([String path = '']) {
  final clean = _cleanSlug(path);
  return clean.isEmpty ? baseUrl : '$baseUrl/$clean/';
}

String homeUrl() => baseUrl;
String skillsUrl() => urlFor('skills');
String submitUrl() => urlFor('submit');
String blogUrl() => urlFor('blog');
String categoryUrl(String slug) => urlFor('category/${_cleanSlug(slug)}');
String skillUrl(String slug) => urlFor('skill/${_cleanSlug(slug)}');
String searchUrl([String query = '']) {
  final value = query.trim();
  return value.isEmpty ? skillsUrl() : '$baseUrl/skills/?q=${Uri.encodeQueryComponent(value)}';
}

File: lib/aiagentskills_site_kit.ex
defmodule AiAgentSkillsSiteKit do
  @moduledoc """
  Small URL helpers for [AI Agent Skills](https://aiagentskills.net).
  """

  @base_url "https://aiagentskills.net"

  def home_url, do: @base_url
  def skills_url, do: url_for("skills")
  def submit_url, do: url_for("submit")
  def blog_url, do: url_for("blog")
  def category_url(slug), do: url_for("category/#{clean_slug(slug)}")
  def skill_url(slug), do: url_for("skill/#{clean_slug(slug)}")

  def search_url(query \\ "") do
    value = String.trim(to_string(query))
    if value == "", do: skills_url(), else: "#{@base_url}/skills/?q=#{URI.encode(value)}"
  end

  def url_for(path \\ "") do
    clean = clean_slug(path)
    if clean == "", do: @base_url, else: "#{@base_url}/#{clean}/"
  end

  defp clean_slug(slug), do: slug |> to_string() |> String.trim() |> String.trim("/")
end
